docs/features/software-catalog/configuration.md

```diff
@@ -192,6 +192,24 @@ Setting this value too low risks exhausting rate limits on external systems that
 are queried by processors, such as version control systems housing catalog-info
 files.
 
+## Pipeline configuration
+
+You can configure the number of entities that are processed in a parallel during the processing loop.
+Both the minimum and maximum amount of entities process can be configured.
+
+- `lowWatermark` defines the target minimum number of items to process in parallel.
+  Once the number of in-flight tasks reaches this count, more tasks will be loaded in.
+- `highWatermark` defines the maximum number of items to process in parallel.
+
+Keep in mind that the `lowWatermark` has to be less than `highWatermark`.
+
+```yaml title="app-config.yaml"
+catalog:
+  pipeline:
+    lowWatermark: 5
+    highWatermark: 10
```

Comment:
Could we make this

```yaml
catalog:
  processing:
    concurrency: { min: 5, max: 10 }
```

instead?

The reason I'm asking is, I've been thinking that it'd be good to move ALL processing loop related things into a `processing` key, and this naming also expresses more clearly what is going on.

Later down the line, we may move `processingInterval` into `processing.interval` etc as well, and maybe have a `processing.enabled` too, but do not do that as part of this PR.

Reply:
done